File: platform/platform-core/fastify/src/routes/platform_health.routes.mjs
/**
 * Platform Health Route
 * GET /health/platform — returns loader reports and diagnostics
 */
export async function mount(server) {
  server.get('/health/platform', async (request, reply) => {
    const reports = server._loaderReports || {};
    const summary = {};

    for (const [name, report] of Object.entries(reports)) {
      summary[name] = {
        discovered: report.discovered ?? report.loaded ?? 0,
        registered: report.registered ?? report.loaded ?? 0,
        skipped: report.skipped ?? 0,
        errors: Array.isArray(report.errors) ? report.errors.length : 0,
      };
    }

    return {
      status: 'ok',
      platform: server._config?.title ?? 'unknown',
      profile: server._config?.profile ?? 'unknown',
      bootTime: server.bootTime ? new Date(server.bootTime).toISOString() : null,
      loaders: summary,
      apps: {
        loaded: server._loadedApps?.length ?? 0,
        skipped: server._skippedApps?.length ?? 0,
      },
    };
  });
}

export default mount;

File: platform/platform-core/fastapi/platform_core_fastapi/routes/platform_health.routes.py
"""Platform Health Route — GET /health/platform returns loader reports and diagnostics."""
from fastapi import Request


def mount(app):
    @app.get("/health/platform")
    async def platform_health(request: Request):
        reports = getattr(request.app.state, 'loader_reports', {})

        summary = {}
        for name, report in reports.items():
            if isinstance(report, dict):
                summary[name] = {
                    'discovered': report.get('discovered', 0),
                    'registered': report.get('registered', 0),
                    'skipped': report.get('skipped', 0),
                    'errors': len(report.get('errors', [])) if isinstance(report.get('errors'), list) else 0,
                }

        config = getattr(request.app.state, 'platform_config', None)
        loaded_apps = getattr(request.app.state, 'loaded_apps', [])
        skipped_apps = getattr(request.app.state, 'skipped_apps', [])

        return {
            'status': 'ok',
            'platform': getattr(config, 'title', 'unknown') if config else 'unknown',
            'profile': getattr(config, 'profile', 'unknown') if config else 'unknown',
            'loaders': summary,
            'apps': {
                'loaded': len(loaded_apps),
                'skipped': len(skipped_apps),
            },
        }
